src/requests.rs

```diff
@@ -29,7 +29,7 @@ use wasm_bindgen::prelude::*;
 /// Publishes end-to-end encryption keys for the device.
 ///
 /// [specification]: https://spec.matrix.org/unstable/client-server-api/#post_matrixclientv3keysupload
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
here and below (x3 overall): is it useful to allow cloning these requests? Not sure how the homeserver will react if we replay them...

Reply:
well, it's useful in that it's a requirement for them to be members of `CrossSigningBootstrapRequests`: the javascript-side getter for `upload_signing_keys_request` needs to be able to clone the `UploadSigningKeysRequest` object. We don't particularly plan to send the request twice, but making it clonable helps fit the rust ownership model to JS semantics.

That said: I now realise that `KeysUploadRequest` is not a member of `CrossSigningBootstrapRequests` (it was in an earlier attempt at this change). Will remove `Clone` for it.

Reply:
```suggestion
#[derive(Debug)]
```